tutorial/Lab2/README.md

```diff
@@ -0,0 +1,155 @@
+    STATUS: DEPLOYED
+    
+    RESOURCES:
+    ==> v1/Service
+    NAME            AGE
+    guestbook-demo  1h
+    redis-master    1h
+    
+    ==> v1/Deployment
+    guestbook-demo  1h
+    redis-master    1h
+    
+    ==> v1/Pod(related)
+    
+    NAME                           READY  STATUS   RESTARTS  AGE
+    guestbook-demo-6c9cf8b9-dhqk9  1/1    Running  0         1h
+    guestbook-demo-6c9cf8b9-zddn2  1/1    Running  0         1h
+    redis-master-5d8b66464f-g7sh6  1/1    Running  0         1h
+    
+    
+    NOTES:
+    Get the application URL by running these commands:
+    export NODE_PORT=$(kubectl get --namespace helm-demo -o jsonpath="{.spec.ports[0].nodePort}" services guestbook-demo)
+    export NODE_IP=$(kubectl get nodes -o jsonpath={.items[*].status.addresses[?\(@.type==\"ExternalIP\"\)].address})
+    echo http://$NODE_IP:$NODE_PORT
+    ```
+    
+    The `upgrade` command upgrades the app to a specified version of a chart, removes the `redis-slave` resources, and updates the app `service.type` to `NodePort`.
+        
+    To check the updates, you can run ```kubectl get all --namespace helm-demo```
+    
+    ```console
+    NAME                                READY     STATUS    RESTARTS   AGE
+    pod/guestbook-demo-6c9cf8b9-dhqk9   1/1       Running   0          1h
+    pod/guestbook-demo-6c9cf8b9-zddn2   1/1       Running   0          1h
+    pod/redis-master-5d8b66464f-g7sh6   1/1       Running   0          1h
+    
+    NAME                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+    service/guestbook-demo   NodePort    172.21.43.244   <none>        3000:31202/TCP   1h
+    service/redis-master     ClusterIP   172.21.12.43    <none>        6379/TCP         1h
+    
+    NAME                             DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+    deployment.apps/guestbook-demo   2         2         2            2           1h
+    deployment.apps/redis-master     1         1         1            1           1h
+    
+    NAME                                      DESIRED   CURRENT   READY     AGE
+    replicaset.apps/guestbook-demo-6c9cf8b9   2         2         2         1h
+    replicaset.apps/redis-master-5d8b66464f   1         1         1         1h
+    ```
+    Note: The service type has changed (to `NodePort`) and a new port has been allocated (`31202` in this output case) to the guestbook service. All `redis-slave` resources have been removed.
+    
+2. View the guestbook as per [Lab1](../Lab1/README.md), using the updated port for the guestbook service.
+
+# Conclusion
+
+Congratulations, you have now updated the applications! Helm does not require any manual changing of resources and is therefore so much easier to upgrade! All configurations can be set on the fly on the command line or by using override files. This is made possible from when the logic was added to the template files, which enables or disables the capability, depending on the flag set.
```

Comment:
Would it be useful to mention that they could easily switch back to the old setup by just modifying the helm cmd? While using kubectl requires finding the old redis yaml files and a lot more work

Reply:
Which you talk about more in the next section - so you could mention these ideas, as a teaser for the next section. Just a thought

Reply:
Can we leave it till Lab3 to cover this? I would prefer to keep it contained in lab 3.

Reply:
Ok, added now